tests/test_tinyaes.py

```diff
@@ -86,3 +111,16 @@ def test_different_keys_do_not_decode(data, aes_enc, aes_dec2, aes_dec3):
     # we can expect a single collision, but not two with two different keys and ivs
     assert aes_dec2.CTR_xcrypt_buffer(encoded) != data \
         or aes_dec3.CTR_xcrypt_buffer(encoded) != data
+
+@given(data=binary(min_size=2, max_size=100))
+def test_different_keys_do_not_decode_cbc(data, aes_enc_cbc, aes_dec2_cbc):
+    data = pad(data)
+    encoded = aes_enc_cbc.AES_CBC_encrypt_buffer(data)
+
+    assert aes_dec2_cbc.AES_CBC_decrypt_buffer(encoded) != data
+
+@given(data=binary(min_size=2, max_size=100))
+def test_cbc_decode(data, aes_enc2_cbc, aes_dec_cbc):
+    data = pad(data)
+    encoded = aes_enc2_cbc.AES_CBC_encrypt_buffer(data)
+    assert aes_dec_cbc.AES_CBC_decrypt_buffer(encoded) == data
```

Comment:
I'm not at all an expert in cryptography, but is `pad()` something like the `PKCS#7` algo suggested in the [tiny-AES-c/README](https://github.com/kokke/tiny-AES-c/tree/f06ac37fc31dfdaca2e0d9bec83f90d5663c319b#tiny-aes-in-c)?

I think in this case the raw binding could follow the wrapped library API, so your code is fine.

Thinking out loud, I'm not sure about the Python naming, should we now expose an `AES_CBC_encrypt_buffer_raw()` to make room for an opinionated `AES_CBC_encrypt_buffer()` that implements the suggested `PKCS#7` padding and decryption in `AES_CBC_decrypt_buffer()`? The ratio is that the main API exposed in Python might be correct and fast in most cases. The `raw` API is exposed for all other needs.

Given the _Important notes:_ in the original `README`, I expect that someone will misuse the API. The problem may be limited by more explicit naming.

Opinions?

Reply:
pad() is an implementation of PKC#7.
As far as I know its the most commonly used padding for AES. its the default padding in OpenSSL.
AES-CTR works without padding. AWS-CBC must have padding. therefore for testing CBC padding must be used.
Regarding the naming, since this is a wrapper I think the APIs should as rule of thumb be named similarly to the wrapped API.
Regarding misusing the API, I think this can be addressed by documentation with examples.
having a fully functional minimal example in the documentation would provide a baseline to anyone who wants to use the APIs.
of course this is just an opinion, I'm not against an additional function that adds padding, but in that case I would suggest validating the padding and removing it in the decryption function.

Reply:
Yes, but in fact we already made the distinction between `_inplace` and copy implementations.

I think a better API for this Python wrapper is to manage the padding internally when the plain `CBC_encrypt_buffer()` name is used.

TL;DR: can you turn your code into an implementation of the `_inplace` version of the methods? Maybe with a check on the length, if possible (see the other comment).

Reply:
changed the CBC functions to be `_inplace` and added block size check and test